components/script/dom/eventtarget.rs

```diff
@@ -375,7 +375,10 @@ impl EventTarget {
             None => self.downcast::<Window>().unwrap().Document(),
         };
 
-        // TODO step 1.2 (browsing context/scripting enabled)
+        // Step 1.2
+        if document.browsing_context().is_none() || !document.is_scripting_enabled() {
```

Comment:
Those two checks are (currently) completely equivalent. Let's just call `is_scripting_enabled()`; that should always include the browsing context check.